### PR TITLE
[telemetry-service] error handling and big query ingest improvements

### LINKED ISSUE #2245: [Aptos Telemetry] Persist telemetry tokens across runs and include role_type.
To help improve telemetry, we'd like to:
1. Persist the telemetry token each node uses to send telemetry data. At the moment, if a node reboots, a new telemetry token is generated in memory. But, this is annoying as it makes it difficult to trace telemetry events across node runs. Instead, we could write the token to local storage and reload it on each reboot (e.g., the same root directory we use for other storage, like the db).
2. We're currently sending the node `role_type` on the `APTOS_NODE_CORE_METRICS` event. But, it sounds like it would make @pr-aptos's life easier if we sent that information on each and every event (to help distinguish between validator and fullnodes sooner).

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

- Service errors are refined and meaningful errors are returned back to the client
- Adding new columns to identify each big query row

Fixes #2245 and #2118

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2996)

### CODE REVIEW

File: crates/aptos-telemetry-service/src/auth.rs
Comment (pr-aptos) [inline crates/aptos-telemetry-service/src/auth.rs:64]: Can it fail here in matching peer_id?  You may want to throw another error if that is the case.

Comment (pr-aptos) [reply crates/aptos-telemetry-service/src/auth.rs:64]: Also, how do we make sure the epoch is the current epoch?

**Review by pr-aptos — APPROVED** [review]
Left a couple of comments for clarification, otherwise looks good from logic pov. If you need, you can wait for a more rust specific code review or merge.

File: crates/aptos-telemetry-service/src/auth.rs
Comment (sitalkedia) [inline crates/aptos-telemetry-service/src/auth.rs:95]: nit: Let's make it more descriptive as the warning message.

```
"Validator set unavailable for Chain ID {}
```

File: crates/aptos-telemetry-service/src/custom_event.rs
Comment (sitalkedia) [inline crates/aptos-telemetry-service/src/custom_event.rs:43]: Let's add the `user_id` and `peer_id` here.

File: crates/aptos-telemetry-service/src/custom_event.rs
Comment (sitalkedia) [inline crates/aptos-telemetry-service/src/custom_event.rs:83]: Let's append `e` to the message.

File: crates/aptos-telemetry-service/src/custom_event.rs
Comment (sitalkedia) [inline crates/aptos-telemetry-service/src/custom_event.rs:99]: same as above

**Review by sitalkedia — APPROVED** [review]
Looks great overall % Minor nits - please address before landing.

File: crates/aptos-telemetry-service/src/custom_event.rs
Comment (ibalajiarun) [reply crates/aptos-telemetry-service/src/custom_event.rs:83]: Hmm,  I am worried if by including the error, we may end up returning any sensitive information regarding our BQ instance. That's why I logged the error before returning a generic error. This may not be helpful to tie those two events together, so I have a [backlog item](https://linear.app/aptoslabs/issue/ADP-90/log-grouping-with-trace-id-for-cloud-run) to address this via tracing support. wdyt?

File: crates/aptos-telemetry-service/src/auth.rs
Comment (ibalajiarun) [reply crates/aptos-telemetry-service/src/auth.rs:64]: Yes, it can fail to get matching peer_id, and we handle it as well in the nested None block.

During authentication, we assume that the validator cache is up-to-date and if we can find the peer in the validator set, then we know its part of that epoch, and generate an auth token with the epoch.

After that, for each request (metrics ingest, big query ingest, etc), we compare the epoch in the auth token to the validator cache epoch and if they differ, we return a 401, and ask client to authenticate again.

File: crates/aptos-telemetry-service/src/custom_event.rs
Comment (sitalkedia) [reply crates/aptos-telemetry-service/src/custom_event.rs:83]: SGTM.

**Pull request merged**
